problem_builder/mentoring.py

```diff
@@ -847,13 +857,19 @@ class MentoringWithExplicitStepsBlock(BaseMentoringBlock, StudioContainerWithNes
         enforce_type=True
     )
 
-    editable_fields = ('display_name',)
+    editable_fields = ('display_name', 'max_attempts', 'extended_feedback')
 
     @lazy
     def questions(self):
         """ Get the usage_ids of all of this XBlock's children that are "Questions" """
         return list(chain.from_iterable(self.runtime.get_block(step_id).steps for step_id in self.steps))
 
+    def get_questions(self):
+        """ Get all questions associated with this block, cached if possible. """
+        if getattr(self, "_questions_cache", None) is None:
+            self._questions_cache = [self.runtime.get_block(question_id) for question_id in self.questions]
+        return self._questions_cache
```

Comment:
@itsjeyd The `questions` method above uses the `@lazy` decorator to cache questions ids. Could `@lazy` be used here as well instead of explicitly caching the blocks?